src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -1103,9 +1103,19 @@ public static void Validate(BoundNode node)
                 return null;
             }
 
-            private void Fail(BoundNode node)
+            public override BoundNode? VisitSequence(BoundSequence node)
             {
-                Debug.Assert(false, $"Bound nodes of kind {node.Kind} should not survive past local rewriting");
+                if (node.ForceSpill)
+                {
+                    Fail(node, extraContent: $" with {nameof(BoundSequence)}.{nameof(node.ForceSpill)} true");
+                }
+
+                return null;
```

Comment:
>return null [](http://example.com/codeflow?start=16&length=11)

It looks like we are adding a new override, not visiting the node here and not delegating to base. Why is this the right thing to do?